server/verify.go

```diff
@@ -293,3 +302,29 @@ func makePool(certs []*x509.Certificate) *x509.CertPool {
 	}
 	return pool
 }
+
+// VerifyTechnology checks Trusted Execution Technology (TEE) attestation reports if the technology
+// supports them, and only then validates that a particular technology has proven that it is in use.
+func VerifyTechnology(attestation *pb.Attestation, tech pb.GCEConfidentialTechnology, opts *VerifyOpts) error {
+	switch tech {
+	case pb.GCEConfidentialTechnology_NONE: // Nothing to verify
+		return nil
+	case pb.GCEConfidentialTechnology_AMD_SEV: // Not verifiable on GCE
+		return nil
+	case pb.GCEConfidentialTechnology_AMD_SEV_ES: // Not verifiable on GCE
+		return nil
+	case pb.GCEConfidentialTechnology_AMD_SEV_SNP:
+		switch tee := attestation.GetTeeAttestation().(type) {
+		case *pb.Attestation_SevSnpAttestation:
```

Comment:
Will there be other possible types for `GCEConfidentialTechnology_AMD_SEV_SNP`? If not, let's just cast and check the result:

```
snpAttest, ok := attestation.GetTeeAttestation.(*pb.Attestation_SevSnpAttestation)
if !ok {
  return fmt.Errorf("TEE attestation is %v, expected an SevSnpAttestation", attestation.GetTeeAttestation())
}
```

Reply:
There will be other technologies, so I could change this into a switch on the type.

Reply:
I understand there will be different `pb.GCEConfidentialTechnology`. I'm wondering if there will be different Attestation types for SevSnp. If not, the above looks a bit cleaner.

Same question for whether there will be different types of Opts for just `pb.Attestation_SevSnpAttestation`. If not let's change it to:
```
snpOpts, ok := opts.TEEOpts.(*VerifySnpOpts)
if !ok {
  return fmt.Errorf("unexpected value for TEEOpts given a SEV-SNP attestation report: %v", opts.TEEOpts)
}
```